### ISSUE TITLE
Add a new property to the autocomplete component to enhance the nonJS functionality

### ISSUE THREAD

**Issue by alexwbbr** [Capgemini/dcx-react-library#657]
A new property needs to be added to the autocomplete component to better handle some nonJS scenarios.
Currently by default if no javascript is detected it will fallback and render a select component with all of the available options, instead of this behaviour a new property needs to be added so that other html elements could be rendered instead.

Please make a new property called nonJsComponent which will take html as a value. If this property is detected and the component isn't hydrated (user has turned off JS) then render the nonJsComponent. If the user hasn't used the new property and the component isn't hydrated then continue it should fallback to the select component by default as we are doing currently.

Please follow these steps to create your branch:
```bash
git checkout release/1.1.0
git pull
git checkout -b 'feature/autocomplete-non-js-fallback'
```

**Comment by daniele-zurico** [comment]
pushed on `feature/autocomplete-non-js-fallback`
the branch created was just to try to experiment with a possible solution (that seems to work as expected). It's missing:
- test
- documentation in storybook

**Comment by daniele-zurico** [comment]
closed by https://github.com/Capgemini/dcx-react-library/pull/661
